modules/modes/rock_smash.py

```diff
@@ -170,6 +170,8 @@ def run(self) -> Generator:
                             break
                         yield
                 case MapRSE.ROUTE121_SAFARI_ZONE_ENTRANCE:
+                    if get_item_bag().quantity_of(get_item_by_name("Pokéblock Case")) == 0:
+                        raise BotModeError("Player does not own the Pokéblock Case yet.")
```

Comment:
There's a feature in rock_smash mode that allows safari re-entry and soft reset to not spew all your cash. I guess it is not used currently in the safari rock smash route.

But I noticed that it would end up in a loop if you don't have the Pokeblock case, because the game doesn't let you enter the safari without it. Probably because there's Pokeblock feeders in the safari